giraffe/src/transforms/annotation.ts

```diff
@@ -0,0 +1,68 @@
+import {AnnotationLayerSpec, AnnotationMark, Table} from '../types'
+import {FILL} from '../constants/columnKeys'
+import {createGroupIDColumn} from './'
+
+export const annotationTransform = (
+  inputTable: Table,
+  inputAnnotations: AnnotationMark[],
+  xColumnKey: string,
+  yColumnKey: string,
+  fillColKeys: string[]
+): AnnotationLayerSpec => {
+  const [fillColumn] = createGroupIDColumn(inputTable, fillColKeys)
+  const table = inputTable.addColumn(FILL, 'system', 'number', fillColumn)
+  const xCol = table.getColumn(xColumnKey, 'number')
+  const yCol = table.getColumn(yColumnKey, 'number')
+
+  let xMin = Infinity
+  let xMax = -Infinity
+  let yMin = Infinity
+  let yMax = -Infinity
+
+  for (let i = 0; i < table.length; i += 1) {
+    if (xCol[i] < xMin) {
+      xMin = xCol[i]
+    }
+
+    if (xCol[i] > xMax) {
+      xMax = xCol[i]
+    }
+
+    if (yCol[i] < yMin) {
+      yMin = yCol[i]
+    }
+
+    if (yCol[i] > yMax) {
+      yMax = yCol[i]
+    }
+  }
+
+  const annotations: AnnotationMark[] = []
+
+  if (Array.isArray(inputAnnotations)) {
+    for (let i = 0; i < inputAnnotations.length; i += 1) {
+      if (
+        (inputAnnotations[i].direction === 'y' &&
```

Comment:
can you add a comment here explaining what these conditions mean? i get what they're saying, but as a reader, i'm not sure what being within these bounds means.

Reply:
Sure. I also changed the second condition to explicitly check for 'x' rather than "not y". Subtle distinction but still a distinction, for the sake of being more explicit.